packages/Animation/FadeAndReveal.jsx

```diff
@@ -8,31 +8,40 @@ export const StyledContainer = styled.div({
     transition: 'none !important',
   },
 })
-const defaultStyle = (duration, timeout, state) => ({
+const defaultStyle = (duration, timeout) => ({
   transition: `height ${duration || timeout}ms, opacity ${duration || timeout}ms ease-in-out`,
   opacity: 0,
   height: 0,
   overflow: 'hidden',
-  visibility: state === 'exited' || state === 'exiting' ? 'hidden' : 'visible',
 })
 
 const transitionStyles = height => (duration, timeout) => ({
-  entering: { opacity: 1, height },
-  entered: { opacity: 1, height: 'auto' },
+  entering: {
+    opacity: 1,
+    height,
+    visibility: 'visible',
+    transition: `height ${duration || timeout}ms, opacity ${duration ||
+      timeout}ms ease-in-out, visibility 0s`,
+  },
+  entered: { opacity: 1, height: 'auto', visibility: 'visible' },
   exiting: {
     opacity: 0,
     height,
-    transition: `height ${duration || timeout}ms, opacity ${duration || timeout}ms ease-in-out`,
   },
-  exited: { opacity: 0 },
+  exited: {
+    opacity: 0,
+    visibility: 'hidden',
+    transition: `height ${duration || timeout}ms, opacity ${duration ||
+      timeout}ms ease-in-out, visibility 0s ${(duration || timeout) / 2}ms`,
```

Comment:
I don't think we need visibility 0s here.

Reply:
Visibility here is necessary because it leverages a delay, which causes it to kick in after the panel has safely collapsed.